records_mover/records/schema/field/__init__.py

```diff
@@ -199,96 +199,104 @@ def components_to_time_str(df: pd.DataFrame) -> datetime.time:
-                          decimal_constraints.fp_significand_bits <= FLOAT64_SIGNIFICAND_BITS):
-                        return np.float64
-                    elif (decimal_constraints.fp_total_bits <= 80 and
-                          decimal_constraints.fp_significand_bits <= FLOAT80_SIGNIFICAND_BITS):
-                        return np.float128
-                    else:
-                        logger.warning("Downgrading float type to np.float128.  "
-                                       "Requested total bits: "
-                                       f"{decimal_constraints.fp_total_bits}.  "
-                                       "Requested significand bits: "
-                                       f"{decimal_constraints.fp_significand_bits}")
-                        return np.float128
-
-            logger.warning(f"No decimal constraints provided for field '{self.name}'; "
-                           "using float64")
-            return np.float64
-        elif self.field_type == 'boolean':
-            return np.bool_
-        elif self.field_type == 'string':
-            return np.object_
-        elif self.field_type == 'date':
-            return np.object_
-        elif self.field_type == 'datetime':
-            return 'datetime64[ns]'
-        elif self.field_type == 'datetimetz':
-            return 'datetime64[ns, UTC]'
-        elif self.field_type == 'time':
-            return np.object_
-        elif self.field_type == 'timetz':
-            return np.object_
-        else:
+                    logger.warning("Downgrading float type to np.float128.  "
+                                   "Requested total bits: "
+                                   f"{decimal_constraints.fp_total_bits}.  "
+                                   "Requested significand bits: "
+                                   f"{decimal_constraints.fp_significand_bits}")
+                    return np.float128
+
+        logger.warning(f"No decimal constraints provided for field '{self.name}'; "
+                       "using float64")
+        return np.float64
+
+    def to_pandas_dtype(self) -> 'Dtype':
+        import numpy as np
+
+        field_type_to_pd_dtype_map = {
+            'integer': self.process_integer_field_type_to_pd_dtype,
+            'decimal': self.process_decimal_field_type_to_pd_dtype,
+            'boolean': np.bool_,
+            'string': np.object_,
+            'date': np.object_,
+            'datetime': 'datetime64[ns]',
+            'datetimetz': 'datetime64[ns, UTC]',
+            'time': np.object_,
+            'timetz': np.object_,
+        }
+
+        pd_dtype = field_type_to_pd_dtype_map.get(self.field_type)
+        if not pd_dtype:
             raise NotImplementedError("Teach me how to handle records schema "
                                       f"type {self.field_type}")
+        if self.field_type in ('integer', 'decimal') and callable(pd_dtype):
```

Comment:
kind of weird to make the map but then have to special case certain types again anyway.

Reply:
Yeah... the other option was to have everything be a lambda again which was kind of a drag. Open to talking through ideas here on Monday!